nncf/common/accuracy_aware_training/__init__.py

```diff
@@ -10,3 +10,33 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
+
+from enum import Enum
+from nncf.config import NNCFConfig
+from nncf.api.compression import CompressionAlgorithmController
+from nncf.config.extractors import extract_accuracy_aware_training_config
+from nncf.common.accuracy_aware_training.training_loop import TrainingLoop
+from nncf.common.accuracy_aware_training.training_loop import EarlyExitCompressionTrainingLoop
+from nncf.common.accuracy_aware_training.training_loop import AdaptiveCompressionTrainingLoop
+
+
+class AccuracyAwareTrainingMode(Enum):
+    EARLY_EXIT = 'early_exit'
+    ADAPTIVE_COMPRESSION_LEVEL = 'adaptive_compression_level'
+
+    @staticmethod
+    def from_str(accuracy_aware_training_mode: str):
+        if accuracy_aware_training_mode == AccuracyAwareTrainingMode.EARLY_EXIT.value:
+            return EarlyExitCompressionTrainingLoop
+        if accuracy_aware_training_mode == AccuracyAwareTrainingMode.ADAPTIVE_COMPRESSION_LEVEL.value:
+            return  AdaptiveCompressionTrainingLoop
+        raise RuntimeError('Incorrect accuracy_aware_training mode')
```

Comment:
It seems very strange to me. Why the `AccuracyAwareTrainingMode` class is responsible for which algorithm will be created? I think that this code
```python
if accuracy_aware_training_mode == AccuracyAwareTrainingMode.EARLY_EXIT:
    return EarlyExitCompressionTrainingLoop
if accuracy_aware_training_mode == AccuracyAwareTrainingMode.ADAPTIVE_COMPRESSION_LEVEL:
    return AdaptiveCompressionTrainingLoop
raise ValueError('Incorrect accuracy_aware_training mode')
```

should be inside of the `create_accuracy_aware_training_loop()` method.

Reply:
Done